README.md

```diff
@@ -3,53 +3,73 @@ dojos
-- inside this copy the `default` directory and name it with the date and your name or pairing name
-- this dir you just copied should have instructions to run the dojo as well as any example code
+Find the module of the dojo you want to attempt. This will be a module under the root dir. Inside you will either find the  `default`  directory or a `java` and `swift` directory.
+
+If you only find the  `default` directory, follow [these instructions](https://github.com/novoda/dojos#Updating-a-dojo). Once the folder structure has been updated properly you can continue, make a copy of the default project of whichever language you want to run the kata with.
+
+Go inside this copy of the `default` directory,  and name it with the date and your name or pairing name. This dir you just copied should have instructions to run the dojo as well as any example code.
 
 ```
 -dojos
 |- theKata
-  |- default
-  |- 01012014-yourname-pairsname
+    |- yourChoosenLanguage
+        |- default
+        |- 01012014-yourname-pairsname
 ```
 
-- add the copied kata to the root `settings.gradle` as `theKata:01012014-yourname-pairsname`
+## Android katas
 
+- add the copied kata to the root `settings.gradle` as `theKata:01012014-yourname-pairsname`
 ```
 include: 'theKata:default', 'theKata:01012014-yourname-pairsname'
 ```
 
-# Running a new dojo
+# Starting a new dojo
 
 - create a new directory inside root with the dojo's name
-- create a `default` directory within which contains the build.gradle file and any template source code to start from
-- a dojo level `build.gradle` containing 
-
-```groovy
-subprojects {
-	apply from: '../default/build.gradle'
-}
-```
-should be added to keep all the implementations using the same build
-
+- create two directories inside the dojo's named `swift` and `java`
+- create a `default` directory inside whichever language you want to run the kata on
+- this `default` directory will contain any template source code and build.graddle file (if you are using a a java project)
 - implementations of the dojo should be at the same directory level as the default directory
- 
+
 ```
 -dojos
 |- myKata
-  |- default
-  |- someonesImpl
+    |- java
+        |- default
+        |- yourImpl
+    |- swift
+        |- default
+        |- yourImpl
 ```
+## Android katas
 
-- add the finished kata's to the root `settings.gradle` as `dojo:implName`
+A dojo level `build.gradle` containing should be added to keep all the implementations using the same build.
+
+```groovy
+subprojects {
+	apply from: '../default/build.gradle'
+}
+```
+Add the finished kata's to the root `settings.gradle` as `dojo:implName`
 
 ```
 include: 'myKata:default'
 ```
 
+# Updating a dojo
 
+If the dojo only contains the `default` folder, it means it has only been used for Android programming and the structure needs to be updated. Create two new folders with  `swift` and `java`  and add everything on the current source dojo folder into java.
```

Comment:
Actually, the majority of the dojos in the repository are not targeted for Android programming. Or they can really only be supported by Android sometimes. E.g kata around a specific library only available in Android.

We should maybe reword this to talk about when we really need to do this. Probably the case would be that the initial dojo was prepared and targeted only to java, then the 2nd language is being added.

Reply:
Yeah wording this was pretty difficult because I do not know all the possible cases and it should be interesting to note possible new ones.  

Answering your comment here I think you are right, I will re write it without being language specific and taking into account that some kata's are for specific language and some have no language specification.

Reply:
Thanks @bertadevant 👍 
My intention was not to block anything. Feel free to shape it however you like. Can also be merged to do a follow-up later. Your call.